src/field.rs

```diff
@@ -7,12 +7,16 @@ use std::fmt;
 
 use num_bigint::BigUint;
 use p3_field::{exp_u64_by_squaring, halve_u32, AbstractField, Field, Packable};
+use rand::{
+  distributions::{Distribution, Standard},
+  Rng,
+};
 use serde::{Deserialize, Serialize};
 
 const PLUTO_FIELD_PRIME: u32 = 101;
-// const MONTY_BITS: u32 = 7;
-// const MONTY_MASK: u32 = (1 << MONTY_BITS) - 1;
-// const MONTY_MU: u32 = 80;
+const MONTY_BITS: u32 = 7;
+const MONTY_MASK: u32 = (1 << MONTY_BITS) - 1;
+const MONTY_MU: u32 = 19; // (-P^-1) mod 2^MONTY_BITS
```

Comment:
Nice work! I think functionally this PR is amazing. My main feedback is i think we should go hard on some comments since we plan to make this a learning resource. If you had to explain to someone what you where doing, who didn't have much prior experience how would you paint the intuition? What intuition made the optimization clique for you? Is there a small example we can include in the comments or another test? Personally i find if i can explain things well it really solidifies my understanding too :)

Reply:
@0xJepsen added comments and examples for montgomery and barret reduction. Let me know if those are not clearly explained.